### DISCUSSION TITLE
flash algorithm for an STM32F423ZHT6

### DISCUSSION THREAD

**Discussion by kasseily** [ARMmbed/DAPLink#1030] [General]
Hello,
I have developed a custom pcb board that has an STM32F103CBT6, and an STM32F423ZHT6.
I want the 103 to act solely as a debugger/programmer to the 423.  They are wired to each other onboard through SWD, and also USART. 
In DAPLink, I am creating a new board for my project, but it seems i need to build a new target_cfg_t structure for the target MCU (in my case the 423). And it needs a flash algorithm, which I do not have.  Does anyone here have this or know where I can find this ?

Thank you

**Comment by carlosperate** [comment]
I'm not familiar with the STM32F4, but you might be able to use the same flash algos for a similar chip and changing the memory regions.
Alternatively, you can look at the Keil website to find flash algos: https://www.keil.arm.com/devices/
And if you cannot find your chip there you can build your own with: https://github.com/pyocd/FlashAlgo

However, it does look like there are algos for this chip: https://www.keil.arm.com/devices/stmicroelectronics-stm32f423zhtx/features/

Download the pack (it's a zip file, you can change the extension) and if the algos come in the `.flm` format you can use this script to generate files that can be used with DAPLink: https://github.com/pyocd/FlashAlgo/blob/main/scripts/generate_blobs.py

**Comment by mathias-arm** [comment]
In DAPLink there we have a custom version of `generate_blobs.py` ([`tools/generate_flash_algo.py`](https://github.com/ARMmbed/DAPLink/blob/main/tools/generate_flash_algo.py)) to convert `.flm` files into the `flash_blob.c` files.

**Comment by kasseily** [comment]
thank you, I will check this out and report back should issues arise.
